### PR TITLE
Better error message when sending to both sprout and sapling

### PR DESCRIPTION
**Pull request by Eirik0**
When trying to send to both Sprout and Sapling (not currently supported with z_sendmany) we were getting the following error in our operation result: `general exception: boost::bad_get: failed value get using boost::get`.

This PR changes this to fail with a better error message before the async operation begins: 
```
error code: -8
error message:
Cannot send to both Sprout and Sapling addresses using z_sendmany
```

### CODE REVIEW

**Comment by mdr0id** [comment]
ACK

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -3710,6 +3713,16 @@ UniValue z_sendmany(const UniValue& params, bool fHelp)
                 bool toSprout = !toSapling;
                 noSproutAddrs = noSproutAddrs && toSapling;
 
+                containsSproutOutput |= toSprout;
+                containsSaplingOutput |= toSapling;
+
+                // Sening to both Sprout and Sapling is currently unsupported using z_sendmany
```
Comment (bitcartel) [inline src/wallet/rpcwallet.cpp]: typo "Sending"

Comment (str4d) [inline src/wallet/rpcwallet.cpp]: Nit: `Sending`

**Comment by mdr0id** [comment]
reACK

**Comment by Eirik0** [comment]
Fixed typo and committed test.

**Comment by bitcartel** [comment]
@zkbot r+

**Comment by daira** [comment]
> [  FAILED  ] WalletTests.ClearNoteWitnessCache (4 ms)

That doesn't look related to this PR, but it may be important.

**Comment by bitcartel** [comment]
Could be transient but here is the error (did not occur on all builders):
'''
[ RUN      ] WalletTests.ClearNoteWitnessCache
wallet/gtest/test_wallet.cpp:1536: Failure
Value of: (bool) saplingWitnesses[1]
  Actual: true
Expected: false
wallet/gtest/test_wallet.cpp:1547: Failure
Value of: (bool) saplingWitnesses[1]
  Actual: true
Expected: false
[  FAILED  ] WalletTests.ClearNoteWitnessCache (4 ms)
'''

**Comment by bitcartel** [comment]
To be investigated but we can proceed with merge since unrelated to that error. @zkbot retry

**Pull request merged**